libs/langgraph/langgraph/pregel/__init__.py

```diff
@@ -1108,7 +1143,13 @@ def put_checkpoint(metadata: CheckpointMetadata) -> Iterator[Any]:
                             del fut, task
 
                     # panic on failure or timeout
-                    _panic_or_proceed(done, inflight, step)
+                    # NOTE: for subgraphs we'll raise GraphInterrupt exception on interrupt
+                    exceptions_to_handle = () if is_subgraph else (GraphInterrupt,)
+                    try:
```

Comment:
can we do an if statement instead? ie don't use a try statement without exceptions to catch